tests/test_cint.py

```diff
@@ -162,3 +203,72 @@ def test_invert_unsigned(ct):
     assert inv_x == y
     assert x == inv_y
 
+
+@pytest.mark.parametrize('ct', FLOATS)
+@pytest.mark.parametrize('op', (mod, lshift, rshift, and_, or_, xor, imod, ilshift, irshift, iand, ior, ixor))
+def test_invalid_operators_on_floats(ct, op):
+    with pytest.raises(TypeError):
+        op(ct(1), ct(1))
+    with pytest.raises(TypeError):
+        op(ct(1), I8(1))    
+    with pytest.raises(TypeError):
+        op(I8(1), ct(1))
+
+@pytest.mark.parametrize('ct', FLOATS)
+def test_int_from_float_conversion(ct):
+    x = ct(1.0)
+    assert int(x) == 1
+    assert ct(int(x)) == x
+
+
+@pytest.mark.parametrize('ct', FLOATS)
+def test_inf_and_nan_values(ct):
+    a = ct("inf")
+    b = ct("nan")
+
+    assert a == float("inf") and math.isinf(a) and a == a
+    assert -a == -float("inf") and math.isinf(-a) and -a == -a
+    assert b != b and math.isnan(b)
+
+@pytest.mark.parametrize('ct', FLOATS)
+@pytest.mark.parametrize('op', (add, sub, mul, truediv, pow))
+def test_inf_and_nan_values_in_operations(ct, op):
+    inf = ct("inf")
+    nan = ct("nan")
+    
+    math.isinf(op(inf, 1))
+    assert math.isinf(op(inf, 1))
+    assert math.isnan(op(nan, 1)) 
+    assert math.isnan(op(2, nan))
+    assert nan ** 0 == 1 ** nan == 1
+
+    assert math.isnan(op(nan, inf))
+    assert math.isnan(op(inf, nan))
+    
+    assert math.isnan(inf / inf)
+    assert math.isnan(inf / -inf)
+    assert math.isnan(inf * 0)
+    assert math.isnan(inf - inf)
+
+    assert math.isinf(op(inf, inf)) or op in (truediv, sub)
+    assert math.isinf(op(inf, -inf)) or op in (truediv, add, pow)
+    
+
+# TODO make this functionality available in the library
+def _test_binary_representation_of_nan_and_inf():
```

Comment:
Use `@pytest.mark.skip(reason="This functionality this test checks for is not implemented yet")` or sth like that

See https://docs.pytest.org/en/6.2.x/skipping.html